From 222932f207c4bafb19860b9e72e9487daea9b390 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Thu, 22 Mar 2012 13:18:52 +0800
Subject: [PATCH 483/678] ENGR00177587 MX6Q_ARM2: add android config for this
 board.

add android config to this board.
only basic boot support.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 Makefile                            |    3 +-
 include/configs/mx6q_arm2_android.h |   83 +++++++++++++++++++++++++++++++++++
 2 files changed, 85 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index b4f953e..05dab52 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-#
+
 # (C) Copyright 2000-2009
 # Wolfgang Denk, DENX Software Engineering, wd@denx.de.
 #
@@ -3311,6 +3311,7 @@ mx6dl_arm2_iram_config		\
 mx6dl_arm2_mfg_config		\
 mx6dl_arm2_lpddr2_config	\
 mx6q_arm2_config		\
+mx6q_arm2_android_config        \
 mx6q_arm2_mfg_config		\
 mx6q_arm2_lpddr2_config		\
 mx6q_arm2_lpddr2pop_config	\
diff --git a/include/configs/mx6q_arm2_android.h b/include/configs/mx6q_arm2_android.h
new file mode 100644
index 0000000..44ed9b6
--- /dev/null
+++ b/include/configs/mx6q_arm2_android.h
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2010-2012 Freescale Semiconductor, Inc.
+ *
+ * Configuration settings for the MX6Q Armadillo2 Freescale board.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ */
+
+#ifndef __ANROID_CONFIG_H
+#define __ANDROID_CONFIG_H
+
+#include "mx6q_arm2.h"
+
+
+/* Disable fastboot and recovery to pass build.
+
+  ARM2 is not a complete android board, only support boot from
+  android. if you want to reference full android functions, fastboot,
+  recovery, please reference sabreSD board which
+  mx6q_sabresd_android.h
+*/
+#if 0
+#define CONFIG_USB_DEVICE
+#define CONFIG_IMX_UDC		       1
+#define CONFIG_FASTBOOT		       1
+#define CONFIG_FASTBOOT_STORAGE_EMMC_SATA
+#define CONFIG_FASTBOOT_VENDOR_ID      0x18d1
+#define CONFIG_FASTBOOT_PRODUCT_ID     0xc01
+#define CONFIG_FASTBOOT_BCD_DEVICE     0x311
+#define CONFIG_FASTBOOT_MANUFACTURER_STR  "Freescale"
+#define CONFIG_FASTBOOT_PRODUCT_NAME_STR "i.mx6q Sabre Smart Device"
+#define CONFIG_FASTBOOT_INTERFACE_STR	 "Android fastboot"
+#define CONFIG_FASTBOOT_CONFIGURATION_STR  "Android fastboot"
+#define CONFIG_FASTBOOT_SERIAL_NUM	"12345"
+#define CONFIG_FASTBOOT_SATA_NO		 0
+#define CONFIG_FASTBOOT_TRANSFER_BUF	0x30000000
+#define CONFIG_FASTBOOT_TRANSFER_BUF_SIZE 0x10000000 /* 256M byte */
+#define CONFIG_ANDROID_RECOVERY
+#endif
+
+#define CONFIG_CMD_BOOTI
+/* which mmc bus is your main storage ? */
+#define CONFIG_ANDROID_MAIN_MMC_BUS 4
+
+#define CONFIG_ANDROID_BOOT_PARTITION_MMC 1
+#define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 5
+#define CONFIG_ANDROID_RECOVERY_PARTITION_MMC 2
+#define CONFIG_ANDROID_CACHE_PARTITION_MMC 6
+
+
+#define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC NULL
+#define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
+	"booti mmc4 recovery"
+#define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
+#define CONFIG_INITRD_TAG
+
+#undef CONFIG_LOADADDR
+#undef CONFIG_RD_LOADADDR
+#undef CONFIG_EXTRA_ENV_SETTINGS
+
+#define CONFIG_LOADADDR		0x10800000	/* loadaddr env var */
+#define CONFIG_RD_LOADADDR      0x11000000
+
+#define CONFIG_INITRD_TAG
+
+#define	CONFIG_EXTRA_ENV_SETTINGS					\
+		"netdev=eth0\0"						\
+		"ethprime=FEC0\0"					\
+		"bootcmd=booti mmc4\0"
+#endif				/* __CONFIG_H */
-- 
1.7.7.4

